extensions/ql-vscode/src/cli.ts

```diff
@@ -485,8 +477,13 @@ export class CodeQLCliServer implements Disposable {
     command: string[],
     commandArgs: string[],
     description: string,
-    cancellationToken?: CancellationToken,
-    logger?: Logger,
+    {
```

Comment:
The aim of the PR is to make things more readable so I'm going to pick on this a bit. Is this definitely more readable? Is it worth defining some type(s) that cover the different kind of options and their default values instead of inlining them in the function definitions, or are the functions just too different?